resources/js/components/QueryBuilder/QueryGroup.vue

```diff
@@ -115,6 +248,54 @@ export default {
 
         removeCondition(conditionIndex) {
             this.$emit('remove-condition', this.groupIndex, conditionIndex);
+        },
+
+        updateNestedGroup(nestedGroupIndex, nestedGroup) {
+            const updatedGroup = { ...this.group };
+            updatedGroup.conditions[nestedGroupIndex] = { ...nestedGroup, type: 'group' };
+            this.$emit('update-group', this.groupIndex, updatedGroup);
+        },
+
+        removeNestedGroup(nestedGroupIndex) {
+            const updatedGroup = { ...this.group };
+            updatedGroup.conditions.splice(nestedGroupIndex, 1);
+            this.$emit('update-group', this.groupIndex, updatedGroup);
+        },
+
+        moveNestedGroupUp(nestedGroupIndex) {
+            if (nestedGroupIndex > 0) {
+                const updatedGroup = { ...this.group };
+                const item = updatedGroup.conditions[nestedGroupIndex];
+                updatedGroup.conditions[nestedGroupIndex] = updatedGroup.conditions[nestedGroupIndex - 1];
+                updatedGroup.conditions[nestedGroupIndex - 1] = item;
+                this.$emit('update-group', this.groupIndex, updatedGroup);
+            }
+        },
+
+        moveNestedGroupDown(nestedGroupIndex) {
+            if (nestedGroupIndex < this.group.conditions.length - 1) {
+                const updatedGroup = { ...this.group };
+                const item = updatedGroup.conditions[nestedGroupIndex];
+                updatedGroup.conditions[nestedGroupIndex] = updatedGroup.conditions[nestedGroupIndex + 1];
+                updatedGroup.conditions[nestedGroupIndex + 1] = item;
+                this.$emit('update-group', this.groupIndex, updatedGroup);
+            }
+        },
+
+        addConditionToNestedGroup(nestedGroupIndex) {
+            this.$emit('add-condition-to-nested', this.groupIndex, nestedGroupIndex);
+        },
+
+        addNestedGroupToNestedGroup(nestedGroupIndex) {
+            this.$emit('add-nested-group-to-nested', this.groupIndex, nestedGroupIndex);
```

Comment:
can't you do these emits in the template right way?

```
@add-nested-group="$emit('add-nested-group-to-nested', groupIndex, nestedGroupIndex)"
```

Or make one function:


emitSomething(event, ..params)

Now you have basicly 4 times the same function that does the same thing but with different params